main.go

```diff
@@ -14,27 +21,81 @@ func main() {
 		log.Fatal(err)
 	}
 
+	if err = run(config); err != nil {
+		config.Logger.Error(errors.Wrap(err, "startup"))
+		os.Exit(1)
+	}
+}
+
+const shutdownTimeoutSeconds = 3
+
+func run(conf *sat.Config) error {
+	localLogger := conf.Logger.CreateScoped("main.run")
+
+	traceProvider, err := sat.SetupTracing(conf.TracerConfig, conf.Logger)
+	if err != nil {
+		return errors.Wrap(err, "setup tracing")
+	}
+	defer traceProvider.Shutdown(context.Background())
+
 	// initialize Reactr, Vektor, and Grav and wrap them in a sat instance
-	s, err := sat.New(config)
+	s, err := sat.New(conf, traceProvider)
 	if err != nil {
-		log.Fatal(err)
+		return errors.Wrap(err, "sat.New")
 	}
 
-	if config.UseStdin {
-		if err := s.ExecFromStdin(); err != nil {
-			log.Fatal(err)
+	if conf.UseStdin {
+		if err = s.ExecFromStdin(); err != nil {
+			return errors.Wrap(err, "sat.ExecFromStdin")
 		}
-
-		return
+		return nil
 	}
 
-	if err := s.Start(); err != nil {
-		if err == http.ErrServerClosed {
-			config.Logger.Info("sat server shutdown complete")
-		} else {
-			config.Logger.Error(errors.Wrap(err, "sat error, dirty shutdown complete"))
+	// Make a channel to listen for an interrupt or terminate signal from the OS. Use a buffered channel because the
+	// signal package requires it.
+	shutdown := make(chan os.Signal, 1)
+	signal.Notify(shutdown, syscall.SIGINT, syscall.SIGTERM)
+
+	// Make a channel to listen for errors coming from the listener. Use a buffered channel so the goroutine can exit if
+	// we don't collect this error.
+	serverErrors := make(chan error, 1)
+
+	go func() {
+		localLogger.Info("startup", "sat with configured vektor, grav, reactr started")
```

Comment:
I don't think we need the `vektor, grav, reactr` part :)

Reply:
Removed in 79fddb29 :)